### ISSUE TITLE
Segmentation fault parsing HTML

### ISSUE THREAD

**Issue by SpyMachine** [ohler55/ox#197]
Hi, I started getting a segmentation fault parsing some code. I tried to investigate the issue, and narrowed it down to this really simple test to get it to reproduce.
```
require 'ox'

html = %q[
<source>
<source>
]

Ox.sax_html(nil, StringIO.new(html), nil)
```
What's interesting is that if I use anything else besides "source" in the tag, it works. Not sure if that's helpful.

Using 2.4.8 btw.

**Comment by ohler55** [comment]
I just tried with the most recent version and get a crash with `source` but nothing else. I'll get it fixed.

**Comment by ohler55** [comment]
If you can move up to the latest I pushed a fix which will be part of the 2.8.2 release. That should be out today or tomorrow.

**Comment by SpyMachine** [comment]
confirmed that fixed the problem, thanks!
